src/commands/clone.js

```diff
@@ -102,13 +112,19 @@ export async function _clone({
       remote,
       noCheckout,
     })
+    return { basename: dirname }
   } catch (err) {
     // Remove partial local repository, see #1283
     // Ignore any error as we are already failing.
     // The catch is necessary so the original error is not masked.
     await fs
       .rmdir(gitdir, { recursive: true, maxRetries: 10 })
       .catch(() => undefined)
+    if (createDir) {
+      await fs
+        .rmdir(dir, { recursive: true, maxRetries: 10 })
+        .catch(() => undefined)
```

Comment:
gitdir is inside dir, why do you need to call rmdir on both directories if it uses `recursive: true` option?

Reply:
You're right, I can make it one rmdir.